ARSIS-Telemetry-GroundControl/groundcontrol-api/app/datastore.py

```diff
@@ -0,0 +1,43 @@
+import asyncio
+import time
+
+from app.tss import get_from_tss, tss_keys
+
+
+class Datastore:
+    def __init__(self):
+        self.new_data_queue = asyncio.Queue(maxsize=500)
+        self.cache = {}
+
+    async def append(self, key, value):
+        if key not in self.cache:
+            self.cache[key] = []
+        self.cache[key].append(value)
+
+    async def start_polling_key(self, key):
+        while True:
+            response = await get_from_tss(key)
+            if response.status_code == 200:
+                new_event = {"type": key, "time": time.time() ,"data": response.json()}
```

Comment:
The event payload given by the WebSocket connection looks like:
```
{
    "type": "telemetry",
    "time": 1707440405.350505,
    "data": {
        "telemetry": {
            "eva_time": 0,
            ...,
        }
    }
}
```

The data still contains the key/type of the data and the time returned in a float value. We would need a way to parse the float data type in Javascript/Typescript and C# into a timestamp. The docs for the time library says to ["Use time_ns() to avoid the precision loss caused by the float type."](https://docs.python.org/3/library/time.html#time.time).

Reply:
Fixed. though the floating point error we'd get would be sub second so I dont think it would matter to much unless for some reason we started summing a lot of times getting the errors to stack.